otsserver/rpc.py

```diff
@@ -198,12 +208,25 @@ def do_GET(self):
             except ZeroDivisionError:
                 time_between_transactions = "N/A"
             transactions.sort(key=lambda x: x["confirmations"])
+
+            lightning_invoice = None
+            lightning_invoice_qr = None
+            if self.lightning_invoice_file is not None:
+                try:
+                    with open(self.lightning_invoice_file, 'r') as file:
+                        lightning_invoice = file.read().strip()
+                        lightning_invoice_qr = get_qr(lightning_invoice)
+                except FileNotFoundError:
+                    pass
```

Comment:
if the file is incomplete the receipt will be partial but the bech32 encoding will recognize it, for corrupt files I can simply catch the broader Exception instead of FileNotFoundError to catch all the cases.
I guess the rename should be done by the process which put the file there, not from here?

Reply:
Failing the bech32 checksum makes sense; catching the broader exception should be fine.

Yeah, the rename needs to be done by the other process. And actually, maybe the way to do this is to delete the file after you've read it *successfully*, cache that qr code, and then check if a *new* file gets put into place?

Reply:
if I delete the file a calendar restart will not find the invoice file which is refreshed only on successful payment...

In the worst case I think bech32 encoding will save us again in the case the scp write the file in the middle of being read by the calendar server, the crap coming out of it will not resemble in a valid invoice.

Reply:
Yeah, that's probably fine then.